### PR TITLE
Issue 424

### PR DESCRIPTION
**Pull request by haleymnatz**
- Configure Invitations controller to allow multiple email addresses to be entered (sends unique token to each user)
- Reconfigure invalid email error on invitations form to be displayed as flash message
- Add input field placeholder with instruction
- Small CSS change to extend field length
- Add contributor blurb
- Tests for invitations controller

Addresses #424.

### CODE REVIEW

**Comment by andy5995** [comment]
Hi @haleymnatz 

I see codeclimate reports 2 errors. Is this a wip?

**Comment by haleymnatz** [comment]
@andy5995 yeah, just 2 rubocop errors to take care of. I'll fix those and merge the latest master into my branch.

File: config/locales/devise_invitable.en.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@ en:
       invited: "You have a pending invitation, accept it to finish creating your account."
     invitations:
       send_instructions: "An invitation email has been sent to %{email}."
+      failed_send: "An error occurred. %{email} could not be invited."
```
Comment (andy5995) [inline config/locales/devise_invitable.en.yml]: From a grammar POV, this sounds a little choppy to me. I'm used to see things like

>Error: ${email} could not be invited

or 

>${email} could not be invited (an unknown error occurred)

But it's not up to me @haleymnatz , I just wanted to make a note of it. What does @julianguyen  or @baohouse think?

**Comment by andy5995** [comment]
You've tested this and it works? If so, I'm confident I can approve it and you can merge it. Looks pretty straightforward. If you've tested it and it works as intended, I'll say LGTM!

File: config/locales/devise_invitable.en.yml (hunk 1 above)
Comment (haleymnatz) [reply config/locales/devise_invitable.en.yml]: @andy5995 I definitely can change it if that's preferred. I'll just have to get the translations for the changed version as well.

**Comment by haleymnatz** [comment]
@andy5995 it passed all of the tests locally, and it sent emails via the dev emailing method (opens up an 'email' tab in my browser). If you're unsure though, it certainly can wait!

**Comment by andy5995** [comment]
Hmmm...  I may as well give Julia a chance to look it over. :) Like you said, no rush. :)

**Comment by julianguyen** [comment]
Tested your branch locally and I'm getting this error:

image

The letter opener gem mocks an email that looks like this:
image

Could you also look into the bug where the subject in the email is %{name}?

File: app/assets/stylesheets/application/allies.scss [hunk 2]
```diff
@@ -12,3 +12,7 @@
 #allies_search {
 	text-align: center;
 }
+
+#user_email {
+	width: 100%;
```
Comment (julianguyen) [inline app/assets/stylesheets/application/allies.scss]: Ooh this is great!

File: app/controllers/users/invitations_controller.rb [hunk 3]
```diff
@@ -3,20 +3,27 @@
 module Users
```
Comment (julianguyen) [inline app/controllers/users/invitations_controller.rb]: This spec doesn't exist currently but should create invitations_controller_spec.rb and tests these methods extensively.

Along with testing the working cases, you should also test invalid input as well!

File: config/locales/devise_invitable.en.yml (hunk 1 above)
Comment (julianguyen) [reply config/locales/devise_invitable.en.yml]: You could just rephrase this to `An error occurred, "%{email}" could not be invited.` I would put quotes between "%{email}" in case there is a blank string.

**Comment by haleymnatz** [comment]
@julianguyen in response to the comment with the screenshot - what functionality would you like to see when the user enters a blank string after a comma? Do you just want me to add quotes around %{email}, or do you want me to have the code ignore the empty space completely and not return an error message?

Also, regarding the %{email} name in the email's subject: that was actually happening on the master branch before I started working on this feature. It was just brought up in another issue ( #752 ). I'd be happy to work on that too, but maybe in a separate PR to keep this one from covering too much?

**Comment by julianguyen** [comment]
@haleymnatz oooh yeah let's fix the subject line bug in another PR then! Good call

**Comment by julianguyen** [comment]
Sorry I didn't reply to the rest of your comment @haleymnatz!

RE:

```
 in response to the comment with the screenshot - what functionality would you like to see when the user enters a blank string after a comma? Do you just want me to add quotes around %{email}, or do you want me to have the code ignore the empty space completely and not return an error message?
```

For both cases, I would always return quotes e.g. `"%{email}"`. Let me know if you have any questions about adding tests for the controller as well! :)

**Comment by haleymnatz** [comment]
@julianguyen I have made the changes your requested and added tests. If you feel that there are any cases/scenarios that I didn't write tests for but should, let me know. Otherwise, I think this is good to go?

**Comment by julianguyen** [comment]
will re-review sometime this weekend! Great work :)

File: app/controllers/users/invitations_controller.rb [hunk 4]
```diff
@@ -3,20 +3,34 @@
...
-        respond_with_navigational(resource) { render :new }
-      end
+    def invitation_flash_messages(invites, fails)
+      return unless is_flashing_format?
+      invites_flash(invites) unless invites.empty?
+      fails_flash(fails) unless fails.empty?
+    end
+
+    def invites_flash(invites)
+      set_flash_message :notice, :send_instructions, email: invites.join(', ')
```
Comment (julianguyen) [inline app/controllers/users/invitations_controller.rb]: Can we join using just `','` (without a space after)?

File: doc/pages/blurbs.json [hunk 5]
```diff
@@ -214,5 +214,14 @@
     "link_name": "maribelduran",
     "link": "http://github.com/maribelduran",
     "social": "github"
+  },
+  {
+    "name": "Haley Mnatzaganian",
```
Comment (julianguyen) [inline doc/pages/blurbs.json]: <3

File: spec/controllers/users/invitations_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            invite_two_friends
+            friend1 = User.find_by(email: invitee1)
+            friend2 = User.find_by(email: invitee2)
+            expect(friend1.invitation_token).not_to eq(friend2.invitation_token)
+          end
+        end
+      end
+
+      context 'when invalid params are given' do
+        let(:invalid_invite) { post :create, params: { user: { email: "#{invalid_email}" } } }
```
Comment (julianguyen) [inline spec/controllers/users/invitations_controller_spec.rb]: I would also test when there's a combination of valid and invalid params e.g. "invalid_email, valid_email@example.com"

File: app/controllers/users/invitations_controller.rb (hunk 3 above)
Comment (julianguyen) [reply app/controllers/users/invitations_controller.rb]: Thanks for addressing this!

File: spec/controllers/users/invitations_controller_spec.rb (hunk 6 above)
Comment (julianguyen) [reply spec/controllers/users/invitations_controller_spec.rb]: Tested locally and it looks like it the valid email will get an invite and the invalid one won't which is good! :)
